training_curve.py

```diff
@@ -0,0 +1,250 @@
+    return [os.path.join(model_dir, name[1]) for name in model_names]
+
+def load_player(model_path):
+  print("Loading weights from %s ... " % model_path)
+  with timer("Loading weights from %s ... " % model_path):
+      network = dual_net.DualNetwork(model_path)
+      network.name = os.path.basename(model_path)
+  player = MCTSPlayer(network, verbosity=2)
+  return player
+
+
+def restore_params(model_path, player):
+  with player.network.sess.graph.as_default():
+    player.network.initialize_weights(model_path)
+
+def batch_run_many(player, positions, batch_size=100):
+  """Used to avoid a memory oveflow issue when running the network
+  on too many positions. TODO: This should be a member function of
+  player.network?"""
+  prob_list = []
+  value_list = []
+  for idx in range(0, len(positions), batch_size):
+    probs, values = player.network.run_many(positions[idx:idx+batch_size])  
+    prob_list.append(probs)
+    value_list.append(values)
+  return np.concatenate(prob_list, axis=0), np.concatenate(value_list, axis=0)
+
+  
+def eval_player(player, positions, moves, results):
+  probs, values = batch_run_many(player, positions)
+  policy_moves = [coords.unflatten_coords(c) for c in np.argmax(probs, axis=1)]
+  top_move_agree = [moves[idx] == policy_moves[idx] for idx in range(len(moves))]
+  square_err = (values - results)**2/4
+  return top_move_agree, square_err
+
+def get_sgf_props(sgf_path):
+  with open(sgf_path) as f:
+    sgf_contents = f.read()
+  collection = sgf.parse(sgf_contents)
+  game = collection.children[0]
+  props = game.root.properties
+  return props
+
+def parse_sgf(sgf_path):
+  with open(sgf_path) as f:
+    sgf_contents = f.read()
+
+  collection = sgf.parse(sgf_contents)
+  game = collection.children[0]
+  props = game.root.properties
+  assert int(sgf_prop(props.get('GM', ['1']))) == 1, "Not a Go SGF!"
+
+  result = utils.parse_game_result(sgf_prop(props.get('RE')))
+
+  positions, moves = zip(*[(p.position, p.next_move) for p in sgf_wrapper.replay_sgf(sgf_contents)])
+  return positions, moves, result, props
+
+def check_year(props, year):
+  if year is None:
+    return True
+  if props.get('DT') is None:
+    return False
+  
+  try:
+    #Most sgf files in this database have dates of the form
+    #"2005-01-15", but there are some rare exceptions like
+    #"Broadcasted on 2005-01-15.
+    year_sgf = int(props.get('DT')[0][:4])
+  except:
+    return False
+  #TODO: better to use datetime comparison here?
+  return year_sgf >= year
+
+def check_komi(props, komi_str):
+  if komi_str is None:
+    return True
+  if props.get('KM') is None:
+    return False
+  return props.get('KM')[0] == komi_str
+
+def find_and_filter_sgf_files(base_dir, min_year = None, komi = None):
+  base_dir = "sgf/baduk_db"
+  sgf_files = []
+  count = 0
+  print("Finding all sgf files in {} with year >= {} and komi = {}".format(base_dir, min_year, komi))
+  for i, (dirpath, dirnames, filenames) in enumerate(os.walk(base_dir)):
```

Comment:
recommend using tqdm for ez progress bars (here and for all for loops that take longer than 10s :)

Reply:
Done